src/Microsoft.Diagnostics.Monitoring.HostingStartup/ParameterCapturing/Pipeline/ParameterCapturingPipeline.cs

```diff
@@ -154,13 +156,36 @@ public bool TryComplete()
             return _requestQueue.Writer.TryComplete();
         }
 
+        private void ValidateMethods(MethodDescription[] methods)
```

Comment:
While this works for this moment, if we ever add fuzzy/partial matching, this gets complicated real quickly. It might be better to move this to after where the actual methods are computed and run this validation on the methods.